File: domain/src/main/java/com/example/labratour/domain/repositories/UserRepository.java
package com.example.labratour.domain.repositories;

import com.example.labratour.domain.Entity.UserDomain;

import io.reactivex.Observable;
import io.reactivex.Single;


//@param userId The user id used to retrieve user data.
public interface UserRepository {

     Observable<UserDomain> login(String email, String  password);
    Observable<String> registerNewUser(String email , String password);
    public Observable<UserDomain> getUser(String userId, boolean fromServer);
     Single<UserDomain> getUser(String userId);

    Observable<Void> saveNewUser(UserDomain userDomain, String userId) ;

}

File: app/src/main/java/com/example/labratour/presentation/model/repositories/UserRepository.kt
package com.example.labratour.presentation.model.repositories

import com.example.labratour.presentation.model.dao.UserDao
import com.example.labratour.presentation.model.data.UserModel

class UserRepository(private val dao: UserDao) {

    suspend fun getUser(id: String): UserModel {
        return dao.getUser(id)
    }

    // insert new user (save)
    suspend fun insertUser(user: UserModel) {
        dao.insertUser(user)
    }

    // delete a spesific user
    suspend fun deleteUser(user_id: String) {
        dao.deleteUser(user_id)
    }
}
